core/src/test/java/org/apache/iceberg/catalog/CatalogTests.java

```diff
@@ -987,6 +991,38 @@ public void testListTables() {
     assertEmpty("Should not contain ns_2.table_1 after drop", catalog, ns2);
   }
 
+  @Test
+  public void emptyNamespace() {
```

Comment:
added this test because JDBC was previously returning namespace `""` (empty string) when listing all namespaces